test-suite/testkit/tests/runtime_api/main.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate assert_matches;
+
+use exonum::{api::Error as ApiError, runtime::rust::ProtoSourceFile};
+use exonum_testkit::{TestKit, TestKitBuilder};
+
+use crate::service::TestRuntimeApiService;
+
+mod proto;
+mod service;
+
+pub fn testkit_with_rust_service() -> TestKit {
+    TestKitBuilder::validator()
+        .with_logger()
+        .with_validators(1)
+        .with_default_rust_service(TestRuntimeApiService)
+        .create()
+}
+
+#[test]
+fn test_exonum_protos_with_service() {
+    let mut testkit = testkit_with_rust_service();
+
+    let response = testkit
+        .api()
+        .public(exonum_testkit::ApiKind::RustRuntime)
+        .get::<Vec<ProtoSourceFile>>("proto-sources");
+
+    match response {
+        Ok(proto_files) => {
+            assert_eq!(proto_files.len(), 7);
+            let proto_names = [
+                "blockchain.proto",
+                "consensus.proto",
+                "doc_tests.proto",
+                "runtime.proto",
+                "tests.proto",
+                "common.proto",
+                "types.proto",
+            ];
+            proto_files
+                .iter()
+                .for_each(|proto| assert!(proto_names.contains(&proto.name.as_ref())));
+        }
+        Err(err) => panic!("Rust runtime Api failed with: {}", err),
```

Comment:
When I told about avoiding `unwrap_err`/`unwrap`, I meant that `assert_eq!(result, Err(some_error)` is better than 

```rust
let error = result.unwrap_err();
assert_eq!(error, some_error);
```

If the comparison within `assert_eq` on `Result` level is not really possible, and `match` is used only for further `panic`, then `expect`/`expect_err` will fit the needs much better.